rpc/blockatlas/jsonrpc_conversion.go

```diff
@@ -0,0 +1,249 @@
+package blockatlas
+
+import (
+	"encoding/hex"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/loomnetwork/go-loom"
+	ltypes "github.com/loomnetwork/go-loom/types"
+	"github.com/pkg/errors"
+)
+
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding
+// Eth JSON RPC define three types QUANTITIES, DATA and default block parameter.
+// All represented by strings.
+type Quantity string
+type Data string
+type BlockHeight string
+
+const (
+	ZeroedQuantity     Quantity = "0x0"
+	ZeroedData         Data     = "0x0"
+	ZeroedData8Bytes   Data     = "0x0000000000000000"
+	ZeroedData20Bytes  Data     = "0x0000000000000000000000000000000000000000"
+	ZeroedData32Bytes  Data     = "0x0000000000000000000000000000000000000000000000000000000000000000"
+	ZeroedData64bytes  Data     = "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+	ZeroedData256Bytes Data     = "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+
+	StatusTxSuccess = "0x1"
+)
+
+type JsonTxObject struct {
+	Hash             Data        `json:"hash,omitempty"`
+	TransactionType  string      `json:"transactionType,omitempty"`
+	ContractName     string      `json:"contractName,omitempty"`
+	ContractMethod   string      `json:"contractMethod,omitempty"`
+	Nonce            Quantity    `json:"nonce,omitempty"`
+	BlockHash        Data        `json:"blockHash,omitempty"`
+	BlockNumber      Quantity    `json:"blockNumber,omitempty"`
+	TransactionIndex Quantity    `json:"transactionIndex,omitempty"`
+	From             string      `json:"from,omitempty"`
+	To               string      `json:"to"`
+	Value            interface{} `json:"value"`
```

Comment:
I still have problems with the type of `Value` field.

What I am trying to do is Decode message and assign the value differently depending on the transaction type 
(Ex `DelegateValue`, `ReDelegateValue` etc.. )

And when I test by query the block that has a trasaction. I got ..
```
{
  "jsonrpc": "2.0",
  "id": "",
  "error": {
    "code": -32603,
    "message": "Internal error",
    "data": "Error marshalling response: Unregistered interface interface {}"
  }
}
```

So Is there a way to add custom struct to these `value` field ?

Reply:
Try using `json.RawMessage` instead of `interface{}` and `json.Marshal` the value before putting it in `JsonTxObject`